lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/util/AzureBatchHelper.java

```diff
@@ -85,6 +96,22 @@ public void submit(final URI jobJarSasUri, final String command) throws BatchErr
     client.jobOperations().createJob(jobAddParameter);
   }
 
+  public List<CloudTask> getTaskStatusForJob(final String jobId) {
+    List<CloudTask> tasks = null;
+    try {
+      tasks = client.taskOperations().listTasks(jobId);
+    } catch (IOException ex) {
+      LOG.log(Level.SEVERE, "Exception when Task status for job: {0}", jobId);
```

Comment:
Let's log the exception here.
Also, this method also throws BatchErrorException which is a subclass of RuntimeException which means it will just get propagated. We should probably catch it here and log it.

Reply:
Done.